google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java

```diff
@@ -977,6 +992,8 @@ public HttpResponse execute() throws IOException {
 
       // execute
       lowLevelHttpRequest.setTimeout(connectTimeout, readTimeout);
+      // switch tracing scope to current span
+      Scope ws = tracer.withSpan(span);
```

Comment:
Before sending the request probably want to record a MessageEvent (network event) with the size of the message sent.